Data/AVL/Tree.idr

```diff
@@ -213,6 +213,50 @@ findKey pred = foldr (\k,v,p => if pred v then Just k else p) Nothing
 findKeyOf : (Eq v) => v -> AVLTree h k v -> Maybe k
 findKeyOf value = findKey (== value)
 
+-- --------------------------------------------------------------------- [ Key ]
+
+||| A proof that some key is found in a Tree
+public export
+data Key : k -> Tree k v -> Type where
+  Here : Key key (Node key _ _ _)
+  InRight : (later : Key key r) -> Key key (Node _ _ _ r)
+  InLeft : (later : Key key l) -> Key key (Node _ _ l _)
+
+||| A wrapper to be used with AVLTree
+public export
+AVLKey : k -> AVLTree h k v -> Type
+AVLKey key avl = Key key (getWitness avl)
+
+||| An empty tree has no key
+noEmptyKey : {key : k} -> Key key Empty -> Void
+noEmptyKey Here impossible
+noEmptyKey (InRight _) impossible
+noEmptyKey (InLeft _) impossible
+
+Uninhabited (Key key Empty) where
+  uninhabited = noEmptyKey
+
+||| An item that is not in the root, not in the left child and not in the
+||| right child is not in the Tree at all
+nowhere : {key : k} -> {val : v} ->
```

Comment:
I am not a fan of the formatting style for type signatures here. Is it possible to reformat this so that arrows are leading rather than trailing?